include/nbl/builtin/hlsl/glsl_compat.hlsl

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_GLSL_COMPAT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_GLSL_COMPAT_INCLUDED_
+
+#include "nbl/builtin/hlsl/spirv_intrinsics.hlsl"
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace glsl
+{
+#pragma region BASIC
+uint gl_SubgroupInvocationID() {
+	// TODO (PentaKon): SPIRV Intrinsics
+	return WaveGetLaneIndex();
+}
+#pragma endregion BASIC
+	
+#pragma region ATOMICS
```

Comment:
suppose you just include these and not use them at all will DXC still emit the SPIR-V for the extensions?

So in a shader stage which does not support subgroup operations, what will happen?

Will Vulkan throw errors left and right?

Reply:
ok you definitely need to split up because of https://github.com/Devsh-Graphics-Programming/Nabla/pull/519/files/51f7dcc8d5a621698bbff96f1e226ea04d4a16eb..e242e96f2d8c7409b9d5345d94158a2ba7810b36#r1279052878